File: Strings/3504/3504.lua
local sentense = io.read()

local ptr1 = 1
local ptr2 = #sentense
local isPalindrome = true
while ptr1 ~= ptr2 do
    if sentense:sub(ptr1, ptr1) ~= sentense:sub(ptr2, ptr2) then
        isPalindrome = false
        break
    end
    ptr1 = ptr1 + 1
    ptr2 = ptr2 - 1
end

if not isPalindrome then 
    print(string.format("A frase [%s] nao eh palindrome", sentense))
end
if isPalindrome then
    print(string.format("A frase [%s] eh palindrome", sentense))
end

File: Strings/3504/3504.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    string sentense;
    getline(cin, sentense);

    size_t ptr1 = 0;
    size_t ptr2 = sentense.size() - 1;
    bool ispalindrome = true;
    while (ptr1 != ptr2) {
        if(sentense[ptr1] != sentense[ptr2]){
            ispalindrome = false;
            break;
        }
        ptr1++;
        ptr2--;
    }

    if (ispalindrome) cout << "A frase [" << sentense << "] eh palindrome\n";
    else cout << "A frase [" << sentense << "] nao eh palindrome\n";

    return 0;
}